internal/pkg/v2/README.md

```diff
@@ -0,0 +1,275 @@
+  - Ancillary supporting structural implementation.
+
+Layers are ordered as they appear (from high -> low): User Interface -> Application -> Domain -> Infrastructure.
+
+The implementation is a relaxed layered architecture.  Higher layers can depend on lower layers.  Lower layers may not depend on higher layers.
+
+
+
+### Project Structure
+
+The intention is to place all APIv2-related code in the `internal/pkg/v2` sub-tree:
+
+```
+internal/pkg/v2						APIv2-related implementation sub-tree
+	README.md							This document
+
+internal/pkg/v2/application			Application Layer
+	delegate							Executable contract delegate implementation
+	dto/v2dot0							Version 2.0 DTO definitions
+	usecases							Use-case implementations
+		common								Common cross-service use-cases
+	validator							DTO validation implementation
+
+internal/pkg/v2/domain				Domain Layer
+
+internal/pkg/v2/infrastructure		Infrastructure Layer
+
+internal/pkg/v2/ui					User Interface Layer
+	common								Transport-independent implementation
+		batchdto							Batch envelope DTO definitions
+		middleware/debugging				Example debugging middleware 
+		routable							Routable delegate implementation
+	http								HTTP transport-specific implementation
+		api									Controller implementation					
+			common								Common cross-service controllers/tests
+			core								Core-service controllers/tests
+			support								Support-service controllers/tests
+			system								System Management controllers/tests
+		handle								Common request handling implementation
+		routing								Version, kind, action router
+```
+
+Note the separation of common cross-service code from service-specific code.
+
+
+
+### Use-case Endpoints
+
+A use-case endpoint is versioned; e.g. `/api/v2/metrics`.
+
+A use-case endpoint accepts either a single use-case request or an array of use-case requests.
+
+A single use-case request returns a single corresponding use-case response.  Success is indicated by an 200 HTTP status code.  The use-case response contains a transport-independent result code that indicates success or a specific failure reason.
+
+An array of use-case requests returns an array of corresponding use-case responses.  A 207 HTTP status code will be returned regardless of the success or failure of the individual requests.  Each use-case response contains a transport-independent result code that indicates success or a specific failure reason.
+
+Multiple use-case requests received in a single transport request must all be of same type (which corresponds to the request DTO expected by the underlying use-case).
+
+Multiple use-case requests received in a single transport request are processed concurrently.  Results are aggregated and returned as a single array of responses in the transport's response.  The relative position of a use-case response in the response's array has no correlation to the relative position of the corresponding use-case request in the request's array.
```

Comment:
IMO, how requests and responses are correlated is an issue for the OpenAPI specification.

Unresolve and comment if you want to continue this conversation here.

Reply:
Since you bring up the correlation issues, i.e. `response's array has no correlation to the relative position` it begs the question on how correlation is resolved.